src/axom/quest/ShapeMesh.cpp

```diff
@@ -0,0 +1,925 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/config.hpp"
+
+#include "axom/primal.hpp"
+#include "axom/quest/ShapeMesh.hpp"
+#include "axom/core/execution/execution_space.hpp"
+#include "axom/fmt.hpp"
+
+#include "conduit_blueprint.hpp"
+
+namespace axom
+{
+namespace quest
+{
+namespace experimental
+{
+#if defined(AXOM_USE_64BIT_INDEXTYPE) && !defined(AXOM_NO_INT64_T)
+static constexpr conduit::DataType::TypeID conduitDataIdOfAxomIndexType = conduit::DataType::INT64_ID;
+#else
+static constexpr conduit::DataType::TypeID conduitDataIdOfAxomIndexType = conduit::DataType::INT32_ID;
+#endif
+
+ShapeMesh::ShapeMesh(RuntimePolicy runtimePolicy,
+                     int allocatorId,
+                     conduit::Node& bpMesh,
+                     const std::string& topoName,
+                     const std::string& matsetName)
+  : m_runtimePolicy(runtimePolicy)
+  , m_allocId(allocatorId != axom::INVALID_ALLOCATOR_ID
+                ? allocatorId
+                : axom::policyToDefaultAllocatorID(runtimePolicy))
+  , m_topoName(topoName.empty() && bpMesh["topologies"].number_of_children() > 0
+                 ? bpMesh["topologies"].child(0).name()
+                 : topoName)
+  , m_matsetName(matsetName.empty() && bpMesh["matsets"].number_of_children() > 0
+                   ? bpMesh.fetch("matsets").child(0).name()
+                   : matsetName)
+  , m_bpNodeExt(&bpMesh)
+  , m_zeroThreshold(1e-10)
+{
+  SLIC_ERROR_IF(m_topoName.empty(),
+                "Topology name was not provided, and no default topology was found.");
+
+  const int hostAllocId = axom::execution_space<axom::SEQ_EXEC>::allocatorID();
+
+  // We want unstructured topo but can accomodate structured.
+  const auto& typeNode =
+    m_bpNodeExt->fetch_existing("topologies").fetch_existing(m_topoName).fetch_existing("type");
+  const std::string topoType = typeNode.as_string();
+  SLIC_ERROR_IF(topoType != "unstructured",
+                "ShapeMesh currently only works with unstructured mesh, not " + topoType + ".");
```

Comment:
This comment (about allowing structured) doesn't match the `SLIC_ERROR`

Reply:
The comment a few lines up says
```cpp
    // We want unstructured topo but can accomodate structured.
```

but this line errors out if it's not "unstructured"
```cpp
   SLIC_ERROR_IF(topoType != "unstructured", "ShapeMesh currently only works with unstructured mesh, not " + topoType + ".");
```